snooty/parser.py

```diff
@@ -1002,6 +1007,23 @@ def validate_tabs_children(self, node: n.Directive) -> None:
         node.children = new_children
         return
 
+    def validate_icon_role(self, node: docutils.nodes.Node) -> None:
+        """
+        Validate target for icon role
+        Checks for included icon file in root path
+        """
+        if not ICON_SET:
+            return
+        icon_name = node.__getitem__("target")
+        possible_name_set = {
```

Comment:
I don't know the answer, just checking: are all of these cases always valid?

Reply:
they're not always valid. ie. there is `fa-archive` and `fa4-archive` but not `archive` so i went about with the set intersection. looking back seems like two prefixes missed ('charts-icon-xxxx' and 'mms-icon-xxxxx')